### ISSUE TITLE
[question] how to replicate a ridig registration done with ANTs / niftyreg

### ISSUE THREAD

**Issue by neuronflow** [SimpleITK/SimpleITK#1966]
I am exploring whether we can replace an ANTs / niftyreg registration step in our pipelines with sitk.
Which parameters would I need to replicate the functionality of:

```
reg_aladin \
    -rigOnly \
    -ref $fixed \
    -flo $moving \
    -res $transformed \
    -aff $matrix \
    -pad 0
```

ANTs we call with the following parameters:
```
if transformationalgorithm == "rigid":
        # rigid ants_transformation
        transform_rigid = "-t rigid[0.1]"
        metric_rigid = "-m Mattes[" + fixed_image + "," + moving_image + ", 1, 32, Regular, 0.5]"
        convergence_rigid = "-c [1000x500x250, 1e-6, 10]"
        smoothing_sigmas_rigid = "-s 3x2x1vox"
        shrink_factors_rigid = "-f 8x4x2"
```

Thanks for your help!

**Comment by zivy** [comment]
Hello @neuronflow,

Most likely this is doable, rigid registration/Mattes Mutual Information/multi-resolution. The ANTs commandline parameters all look very similar to the settings in SimpleITK. Please take a look at [this jupyter notebook](https://github.com/InsightSoftwareConsortium/SimpleITK-Notebooks/blob/master/Python/61_Registration_Introduction_Continued.ipynb). Specifically, the **Initial Alignment** and **Version 2** of the final registration.

**Comment by neuronflow** [comment]
Thanks for the quick response. In our case, we are trying to register 3D brain MR across modalities. If we understood you correctly we should use:
```
initialize: geometry
transform: euler
metric: mattes
```

**Comment by zivy** [comment]
The specifics listed are trying to mimic the settings you are using with the ANTs command-line. You will probably have to modify the settings slightly to account for differences with ITK/SimpleITK, but generally speaking all the code you need is in the referenced notebook. Specifically:
* `transform_rigid = "-t rigid[0.1]"` maps to `Euler3DTransform`
* `metric_rigid = "-m Mattes[" + fixed_image + "," + moving_image + ", 1, 32, Regular, 0.5]` maps to `SetMetricAsMattesMutualInformation`, not sure what the parameters are, but they likely match the settings for the MI metric (number of histogram bins, sampling strategy and sampling percentage).
* `convergence_rigid = "-c [1000x500x250, 1e-6, 10]"` maps to settings for `SetOptimizerAsGradientDescent` (possibly some other optimizer, but try this one for a start).
* `smoothing_sigmas_rigid = "-s 3x2x1vox"` maps to `SetSmoothingSigmasPerLevel`
* `shrink_factors_rigid = "-f 8x4x2"` maps to `SetShrinkFactorsPerLevel`

The only other parameter that isn't part of the registration is `-pad 0` which is likely used by the resampling, after registration. Use zero as the value outside the image. That is the default behavior in the SimpleITK resampling code, so no need to specify unless you want to change it (see docs for [resampling class](https://simpleitk.org/doxygen/latest/html/classitk_1_1simple_1_1ResampleImageFilter.html) and [procedural interface](https://simpleitk.org/doxygen/latest/html/namespaceitk_1_1simple.html#aadbb243c10d1aedea8e955e8beda4df0)).

Also, the initialization is not specified so likely it is "geometry" which is one of the more straightforward initialization approaches. Registration initialization is **critical** for success, see [this notebook](https://github.com/InsightSoftwareConsortium/SimpleITK-Notebooks/blob/master/Python/63_Registration_Initialization.ipynb) for discussion.

**Comment by neuronflow** [comment]
Thank you so much!

You are right, Apologies, I failed to provide the whole call. Here is the rest we have in place:

```
    dimensionality = "-d 3"
    initial_moving_transform = "-r [" + fixed_image + ", " + moving_image + ", 0]"
    
    ..
    
    use_estimate_learning_rate_once = "-l 1"
    collapse_output_transforms = "-z 1"
    interpolation = "-n BSpline[3]"
    precision = "--float 1"
    output = "-o " + "[" + outputmat + "]"
    ```
